### PR TITLE
feat: add script for building rocks via lpci

### PR DESCRIPTION
**Pull request by cjdcordeiro**
In this PR there is a new standalone Python script that takes a `rockcraft.yaml` file from the current directory and offloads the corresponding builds to Launchpad, via lpci.

# How does it work?

1. the user provides a valid LP credential (or creates one during the first run of the script, interactively)
2. a local Git repo is created in a temporary folder, and all the contents from the user's current directory are copied there
3. a `.launchpad.yaml` configuration file is created in this temporary git repo. The yaml file is populated according to the `rockcraft.yaml`'s `build-base` and `platforms`
4. a brand new repo (with a time-based identifier) is created in Launchpad
5. the local temporary git repo, with the `.launchpad.yaml` file, is pushed to the newly created LP repo
6. the above step automatically triggers the CI builds in Launchpad
7. the script keeps fetching those builds' statuses until they are all complete
8. finally, for the successful builds, the rock artefacts (aka OCI archives) are downloaded into the current directory

# Try it

To try it, you can simply

```bash
pip install -r requirements.txt
./requirements.sh
rockcraft init   # to get a demo rockfile to play with
./rockcraft_lpci_build.py --lp-credentials-file <path-to-where-you-want-to-save-your-creds> --allow-build-failures 
```

### CODE REVIEW

**Comment by linostar** [comment]
Add a README with the following steps:

```
How to use:

1. Clone the repo and go to rockcraft_lpci repo. Optionally, create a virtualenv dir there and activate the virtualenv
2. Run: pip install -r requirements.txt
3. Run: ./requirements.sh
4. Create/move your rockcraft.yaml in the current dir
5. Run: python3 rockcraft_lcpi_build.py --allow-build-failures --lp-credentials-file <path-to-where-you-want-to-save-your-creds> 
```

**Comment by linostar** [comment]
Is the `--lp-credentials-b64` argument necessary? It apparently requires the oauth token/secret encoded in base64, but it doesn't tell the user that explicitly, plus the user has to create the oauth token manually.

**Comment by linostar** [comment]
I am not sure if this is a desired effect, but I'll mentioned it here.
Whenever a build fails, launchpad sends an email notification containing a link to the build log. However, if you are not fast enough the build logs link will give 404 because the temporary repo will be deleted once all builds are done even if they fail.

File: rockcraft_lpci_build/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+distro-info
+GitPython
+launchpadlib
+pyyaml
+retry
```
[the remark this answers is not in the document]
Comment (linostar) [reply rockcraft_lpci_build/requirements.txt:5]: Must be an overlook, nvm it.

**Review by linostar — APPROVED** [review]
Looks good to go.

**Comment by cjdcordeiro** [comment]
@linostar this is ready for review. the unit tests are still very minimal, but we might just go ahead with them like this just to get something started for now

**Review by linostar — APPROVED** [review]
A wholesome effort. Great work.

**Pull request merged**